liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -148,6 +157,16 @@ public Database getDatabase() {
         return database;
     }
 
+    /**
+     * Returns the Database used by this Liquibase instance for locking.
+     * <p>
+     * Should be a separate one as some lock services might commit transactions asynchronously,
+     * and hence should not rollback or commit what the actual updates is doing.
```

Comment:
This is a separate *connection* to the primary database, not a separate database (in the normal sense of the term) for locking, right?

Are you sure this is needed? All the changelogs operate in their own transactions, so the current `master` version of Liquibase is able to use the same connection for the lock operations as for the main operations. What has changed in that regard to make this necessary?

The "database" terminology here is confusing. It would be a lot clearer to use a separate connection to the primary database (if indeed it is necessary), rather than have it labelled as a separate "database" all over the place. Your hands might be tied here if the Liquibase "database" object actually represents a connection rather than a datbase, but I think Liquibase's wrappers do distinguish between a database and a connection.

It also seems like a bug magnet: if any future changes forget to use the correct "database", they will introduce some rather subtle bugs.

Reply:
Yeah. You are totally right about the wording, but if I remember correctly, the liquibase "Database" is what I would call a "Connection" in general. I did not find a "Connection" Object in Liquibase that can be used for this purpose, do you know if that exists?

In fact, the separate connection is needed. My lock service actively prolongs the lock in the background (writing a new timestamp every x seconds), and in order to make this change visible to other clients, it needs to commit the transaction every x seconds, but of course you would not want the actual updates to the database schema (that are still in progress and incomplete) get committed when this happens.

Reply:
True this could be a bug magnet, do you have a suggestion how to improve that?